common/src/main/java/org/apache/sedona/common/raster/RasterBandAccessors.java

```diff
@@ -78,6 +80,68 @@ public static long getCount(GridCoverage2D raster, int band) {
 //        return getCount(raster, 1, excludeNoDataValue);
 //    }
 
+    public static double[] getSummaryStats(GridCoverage2D rasterGeom, int band, boolean excludeNoDataValue) {
+        ensureBand(rasterGeom, band);
+        Raster raster = rasterGeom.getRenderedImage().getData();
+        int height = RasterAccessors.getHeight(rasterGeom), width = RasterAccessors.getWidth(rasterGeom);
+        double[] pixels = raster.getSamples(0, 0, width, height, band - 1, (double[]) null);
+        double count = 0, sum = 0, mean = 0, stddev = 0, min = Double.MAX_VALUE, max = -Double.MAX_VALUE;
+        if(excludeNoDataValue) {
+            // exclude no data values
+            Double noDataValue = RasterBandAccessors.getBandNoDataValue(rasterGeom, band);
+            for (double pixel: pixels) {
+                if (noDataValue == null || pixel != noDataValue) {
+                    count++;
+                    sum += pixel;
+                    if (pixel < min) {
+                        min = pixel;
+                    }
+                    if (pixel > max) {
+                        max = pixel;
+                    }
+                }
+            }
+            mean = sum / count;
```

Comment:
We need to be aware of the cases when all pixels in the band are nodata value. PostGIS returns all statistics except count as `NULL`. I think both NULL and NaN are sane choices.

Reply:
Okay, will add a check before the compute code and return NULL.